### PR TITLE
servers: allow building with FFT_GREEN on macOS

### PR DESCRIPTION
**Pull request by dyfer**
## Purpose and Motivation

Previously, CMake option `FFT_GREEN` did not work on macOS (regardless of its value, one would always build against the `vDSP` library). This PR allows using this built-in FFT library on macOS.

The default value stays the same (`FFT_GREEN=OFF`) since it is not recommended.

## Types of changes

- Bug fix

## To-do list

- [x] Code is tested
- [x] This PR is ready for review

### CODE REVIEW

File: server/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,13 @@ include_directories(${boost_include_dirs})
 
 # here we choose who provides us with the FFT lib
 if (APPLE)
-	add_definitions("-DSC_FFT_VDSP")
+	if (FFT_GREEN)
+		message(STATUS "Using green fft")
+		set(FFT_GREEN 1)
```
Comment (brianlheim) [inline server/CMakeLists.txt]: why is this necessary? we already got here with FFT_GREEN equal to a CMake-truthy value

**Comment by dyfer** [comment]
> Thanks! Curious, I don't know much about FFT libraries, is there any reason someone might conceivably want to use this?

My understanding is that it's the obsolete library, potentially useful when debugging issues with other FFT libraries or FFT in general. I just used it to [investigate an issue here](https://github.com/supercollider/supercollider/issues/4289#issuecomment-532006937).

File: server/CMakeLists.txt (hunk 1 above)
Comment (dyfer) [reply server/CMakeLists.txt]: You're right, this is a copy-paste from the code below. I'll remove that.

**Comment by sonoro1234** [comment]
why not FFTW?

**Comment by dyfer** [comment]
> why not FFTW?

This issue is about enabling FFT_GREEN. Do you suggest I should expand it to allow FFTW on macOS?

**Comment by sonoro1234** [comment]
https://github.com/supercollider/supercollider/issues/4289#issuecomment-532366877

**Comment by dyfer** [comment]
I will likely work on enabling the use of FFTW on macOS for supernova, which would supersede this PR. Closing for now.
